petl/io/remotes.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, division
+
+import io
+import logging
+from contextlib import contextmanager
+
+from petl.compat import PY3
+from petl.io.sources import register_reader, register_writer
+
+logger = logging.getLogger(__name__)
+
+# region RemoteSource
+
+
+class RemoteSource(object):
+    """Read or write directly from files in remote filesystems.
+    
+    This source handles many filesystems that are selected based on the
+    protocol passed in the `url` argument.
+    
+    The url should be specified in `to..()` and `from...()` functions. E.g.::
+
+        >>> import petl as etl
+        >>> 
+        >>> def example_s3():
+        ...     url = 's3://mybucket/prefix/to/myfilename.csv'
+        ...     data = b'foo,bar\\na,1\\nb,2\\nc,2\\n'
+        ... 
+        ...     etl.tocsv(data, url)
+        ...     tbl = etl.fromcsv(url)
+        ...     
+        >>> example_s3() # doctest: +SKIP
+        +-----+-----+
+        | foo | bar |
+        +=====+=====+
+        | 'a' | '1' |
+        +-----+-----+
+        | 'b' | '2' |
+        +-----+-----+
+        | 'c' | '2' |
+        +-----+-----+
+
+    This source uses `fsspec`_ to provide the data transfer with the remote
+    filesystem. Check the `Built-in Implementations <fs_builtin>`_ for available
+    remote implementations.
+
+    Some filesystem can use `URL chaining <fs_chain>`_ for compound I /O.
+
+    .. note::
+
+        For working this source require `fsspec`_ to be installed, e.g.::
+
+            $ pip install fsspec
+
+        Some remote filesystems require aditional packages to be installed.
+        Check  `Known Implementations <fs_known>`_ for checking what packages
+        need to be installed, e.g.::
+
+            $ pip install s3fs     # AWS S3
+            $ pip install gcsfs    # Google Cloud Storage
+            $ pip install abfs     # Azure Blob service
```

Comment:
```suggestion
            $ pip install adlfs    # Azure Blob service
```

It looks like Azure support is provided by the [adlfs](https://github.com/dask/adlfs) package.